independent-projects/bootstrap/runner/src/main/java/io/quarkus/bootstrap/runner/RunnerClassLoader.java

```diff
@@ -101,18 +105,48 @@ public Class<?> loadClass(String name, boolean resolve) throws ClassNotFoundExce
                     continue;
                 }
                 definePackage(packageName, resources);
-                try {
-                    return defineClass(name, data, 0, data.length, resource.getProtectionDomain());
-                } catch (LinkageError e) {
-                    loaded = findLoadedClass(name);
-                    if (loaded != null) {
-                        return loaded;
+                return defineClass(name, data, resource);
+            }
+        }
+        return getParent().loadClass(name);
+    }
+
+    private void definePackage(String pkgName, ClassLoadingResource[] resources) {
+        if ((pkgName != null) && getDefinedPackage(pkgName) == null) {
+            for (ClassLoadingResource classPathElement : resources) {
+                ManifestInfo mf = classPathElement.getManifestInfo();
+                if (mf != null) {
+                    try {
+                        definePackage(pkgName, mf.getSpecTitle(),
+                                mf.getSpecVersion(),
+                                mf.getSpecVendor(),
+                                mf.getImplTitle(),
+                                mf.getImplVersion(),
+                                mf.getImplVendor(), null);
+                    } catch (IllegalArgumentException e) {
+                        var loaded = getDefinedPackage(pkgName);
+                        if (loaded == null) {
+                            throw e;
+                        }
                     }
-                    throw e;
+                    return;
                 }
             }
+            definePackage(pkgName, null, null, null, null, null, null, null);
```

Comment:
This one also needs a `try`/`catch` check just like above.

Reply:
Oops, fixed